PlayerCore/components/AdFinishTracker.swift

```diff
@@ -4,19 +4,22 @@
 import Foundation
 
 public struct AdFinishTracker {
-    public let isFinished: Bool
+    public let isForceFinished: Bool
+    public let isSuccessfullyCompleted: Bool
 }
 
 func reduce(state: AdFinishTracker, action: Action) -> AdFinishTracker {
     switch action {
     
-    case is ShowAd, is AdMaxShowTimeout:
-        return AdFinishTracker(isFinished: false)
+    case is ShowAd:
+        return AdFinishTracker(isForceFinished: false, isSuccessfullyCompleted: false)
```

Comment:
What the meaning of `forceFinish`

Reply:
Means that the ad wasn't completed due to some reason (max show timeout, skip, stop etc.)